core/optaplanner-core-impl/pom.xml

```diff
@@ -131,6 +131,10 @@
           </sources>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>animal-sniffer-maven-plugin</artifactId>
+      </plugin>
```

Comment:
Why only `core-impl`? I'd argue we want this in every module, unless we can prove that the module will not work on Android. (Drools does not work on Android, for example.)

Reply:
True, I used it to check if it truly will catch the api incompatibility that was fixed by e5f33e795e9ee013756f99e7c573674142321666

Reply:
Fixed

Reply:
I would add also the `optaplanner-persistence` module to make sure we can convert/persist score on Android.